website/project/views/register.py

```diff
@@ -151,10 +424,32 @@ def node_register_template_page_post(auth, node, **kwargs):
     schema = MetaSchema.find(
         Q('name', 'eq', template)
     ).sort('-schema_version')[0]
+
+    # Create the registration
     register = node.register_node(
         schema, auth, template, json.dumps(clean_data),
     )
 
+    if data['registrationChoice'] == 'Make registration public immediately':
+        register.is_public = True
+        register.save()
+    elif data['registrationChoice'] == 'Enter registration into embargo':
+        # Sanitize embargo end date
+        embargo_end_date = datetime.datetime.strptime(
+            data['embargoEndDate'],
+            "%a, %d %b %Y %H:%M:%S %Z"
+        ).date()
+
+        # Initiate embargo
+        try:
+            register.embargo_registration(auth.user, embargo_end_date)
```

Comment:
There is no need to pass a string to `embargo_registration`. The method should take a `datetime` object.